src/Vira/Supervisor/Type.hs

```diff
@@ -32,3 +32,9 @@ data Task = Task
   -- ^ The `Async` handle for the task
   }
   deriving stock (Generic)
+
+-- | Exceptions thrown during a task
+data TaskException = UserKilled
+  deriving stock (Show)
```

Comment:
Documentation is incomplete. A casual reader won't know what `UserKilled` means (which name, btw, is strange since it reads as "user [got] killed").

Reply:
Renamed to `KilledByUser` and added documentation in https://github.com/juspay/vira/pull/50/commits/1dcef2ee2f7d152dd4a2b9032976197c44fc5d58